apis/handlers/addprog.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Contributors to the L3AF Project.
+// SPDX-License-Identifier: Apache-2.0
+
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/l3af-project/l3afd/kf"
+	"github.com/l3af-project/l3afd/models"
+)
+
+// AddEbpfPrograms add new eBPF programs on node
+// @Summary Adding new eBPF Programs on node
+// @Description Adding new eBPF Programs on node
+// @Accept  json
+// @Produce  json
+// @Param cfgs body []models.L3afBPFPrograms true "BPF programs"
+// @Success 200
+// @Router /l3af/configs/v1/add [post]
+func AddEbpfPrograms(ctx context.Context, kfcfg *kf.NFConfigs) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		mesg := ""
+		statusCode := http.StatusOK
+
+		w.Header().Add("Content-Type", "application/json")
+
+		defer func(mesg *string, statusCode *int) {
+			w.WriteHeader(*statusCode)
+			_, err := w.Write([]byte(*mesg))
```

Comment:
If we cannot write, should we continue to limp along? I don't quite understand what would happen here after it logs the message?

Reply:
just return from function because  defer functions will execute in the end of parent function